exp rename: Add rename command (#4805)
iterative/dvc.org: content/docs/command-reference/exp/rename.md

# exp rename

Rename a specific experiment from the <abbr>project</abbr>.

> 

Reviewer: cc @dberenbaum

We should probably add an info admon that this is a new command supported as of version 3.x.x
What's our convention here? I couldn't find an example

Reviewer 2: We generally don't do this. I'm not against it but it will definitely generate opinions, so let's not let it block this PR.

Author: P.s. as a user it would be nice to have this in all docs (in my opinion). But like @dberenbaum said, this is not the standard, so I haven't added it here.

## Synopsis

```usage
usage: dvc exp rename [-h] [-q | -v] [--rev <commit>]
                      [-g <git_remote>]
                      [experiment] [name]

positional arguments:

>    experiment    Experiments to remove.

Reviewer: ```suggestion
   experiment    Experiments to rename.
```

   name          Name of new experiment.
```

## Description

Renames one experiment, indicated by name (see `dvc exp run`) or rev.

## Options

> - `--rev <commit>` - rename experiments derived from the specified `<commit>` as

Reviewer: ```suggestion
- `--rev <commit>` - rename all experiments derived from the specified `<commit>` as
```
Does that make sense? multiple experiments with the same name 🤔

Reviewer 2: I believe it should only rename one experiment, but experiment names are only unique by `rev`, so you may need to specify the `rev` to rename an experiment from any commit other than the current HEAD. @BradyJ27 If that's the case, can we clarify the default behavior?

Author: Removed `--rev` flag in general.

> - `-g`, `--git-remote` - Name or URL of the Git remote to rename the experiment

Reviewer 3: minor: Name -> name

>   from

Reviewer 3: from.

- `-g`, `--git-remote` - Name or URL of the Git remote to rename the experiment
  from

- `-h`, `--help` - shows the help message and exit.

- `-q`, `--quiet` - do not write anything to standard output. Exit with 0 if no
  problems arise, otherwise 1.

- `-v`, `--verbose` - displays detailed tracing information.

## Examples

> Let's say we have `dvc exp run` 3 experiments in our project:

Reviewer: ```suggestion
Let's say we have created 3 experiments in our project using `dvc exp run`:
```

```cli
$ dvc exp list
master:
        major-mela
        conic-ease
        lucid-lair
```

> To rename any of them, give their names to `dvc exp rename`:

Reviewer: ```suggestion
To rename any of them, pass the original and new experiment names to `dvc exp rename`:
```

```cli
$ dvc exp rename conic-ease renamed-exp

$ dvc exp list
master:
        major-mela
        renamed-exp
        lucid-lair
```

> The experiment named "conic-ease" has now been renamed to "renamed-exp" as seen

Reviewer: ```suggestion
The experiment named "conic-ease" has now been renamed to "renamed-exp".
```

from the output of `dvc exp list`.

> We can also remove experiments from a remote Git repository:

Reviewer: ```suggestion
We can also rename an experiment from a remote Git repository:
```
